tiledb/sm/storage_manager/storage_manager.cc

```diff
@@ -1495,21 +1529,20 @@ Status StorageManager::get_fragment_info(
   fragment_info->set_dim_info(
       array_schema->dim_names(), array_schema->dim_types());
 
-  std::vector<tdb_shared_ptr<FragmentMetadata>> fragment_metadata;
-
   // Get encryption key from config
-  RETURN_NOT_OK(array_reopen(
-      array.array_uri(),
-      *array.encryption_key(),
-      &array_schema,
-      &fragment_metadata,
-      array_type == ArrayType::SPARSE ? timestamp_start : 0,
-      timestamp_end));
+  auto&& [st, array_schema_opt, array_schemas_opt, fragment_metadata_opt] =
+      array_reopen(
+          array.array_uri(),
+          *array.encryption_key(),
+          array_type == ArrayType::SPARSE ? timestamp_start : 0,
+          timestamp_end);
 
   // Return if array is empty
-  if (fragment_metadata.empty())
+  if (fragment_metadata_opt.value().empty())
```

Comment:
Needs guarding by a condition that ensures that the `optional` has a value. The `RETURN_NOT_OK` macro was not replace with a status test. It would be sufficient to rely on the postcondition instead of also testing with `has_value`.